### PR TITLE
Ubuntu 22.04 kernel 6.2 Changes

### PR DESCRIPTION
**Pull request by Bhyaswanth**
Added support for 6.2 kernel.

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c [hunk 1]
```diff
@@ -4097,6 +4097,9 @@ MODULE_AUTHOR("Xilinx, Inc.");
 MODULE_DESCRIPTION(DRV_MODULE_DESC);
 MODULE_VERSION(DRV_MODULE_VERSION);
 MODULE_LICENSE("GPL v2");
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,16,0) || defined(RHEL_9_0_GE)
```
Comment (sureshreddyavula) [inline src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c:4100]: @Bhyaswanth Why do you need to import DMA_BUF? If you remove this, do you see any errors?

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.c [hunk 2]
```diff
@@ -1230,10 +1231,27 @@ int xocl_gem_prime_vmap(struct drm_gem_object *obj, struct dma_buf_map *map)
...
+}
+
+void xocl_gem_prime_vunmap(struct drm_gem_object *obj, struct iosys_map *map)
+{
+
 }
 #endif
 
 
+
```
Comment (sureshreddyavula) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.c]: Please remove the additional line

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 3]
```diff
@@ -1890,3 +1894,6 @@ MODULE_VERSION(XRT_DRIVER_VERSION);
 MODULE_DESCRIPTION(XOCL_DRIVER_DESC);
 MODULE_AUTHOR("Lizhi Hou <lizhi.hou@xilinx.com>");
 MODULE_LICENSE("GPL v2");
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,16,0)  || defined(RHEL_9_0_GE)
+MODULE_IMPORT_NS(DMA_BUF);
```
Comment (sureshreddyavula) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c:1898]: Same question here for importing DMA_BUF

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 4]
```diff
@@ -405,7 +405,12 @@ zocl_gem_mmap(struct file *filp, struct vm_area_struct *vma)
 	 * and set the vm_pgoff (used as a fake buffer offset by DRM)
 	 * to 0 as we want to map the whole buffer.
 	 */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0)
 	vma->vm_flags &= ~VM_PFNMAP;
+#else
+    vm_flags_clear(vma, VM_PFNMAP);
```
Comment (sureshreddyavula) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c]: Use tab

File: src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c (hunk 1 above)
Comment (Bhyashwanth) [reply src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c:4100]: Yes @sureshreddyavula , We are seeing the DKMS configure issue while installing XRT on 6.2 kernel.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c (hunk 3 above)
Comment (Bhyashwanth) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c:1898]: Yes @sureshreddyavula , We are seeing the DKMS configure issue while installing XRT on 6.2 kernel.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.c (hunk 2 above)
Comment (Bhyashwanth) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.c]: Sure.

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c (hunk 4 above)
Comment (Bhyashwanth) [reply src/runtime_src/core/edge/drm/zocl/zocl_drv.c]: Okay

**Comment by Bhyaswanth** [comment]
@sureshreddyavula Hi suresh I have pushed the latest commit with the required changes.

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -62,7 +62,7 @@ SET(XRT_VERSION_MINOR 11)
 if (DEFINED ENV{XRT_VERSION_PATCH})
   SET(XRT_VERSION_PATCH $ENV{XRT_VERSION_PATCH})
 else(DEFINED $ENV{XRT_VERSION_PATCH})
-  SET(XRT_VERSION_PATCH 0)
+  SET(XRT_VERSION_PATCH 722)
```
Comment (rbramand-xilinx) [inline src/CMakeLists.txt:65]: This change should be removed when sending PR to XRT repo

File: src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c (hunk 1 above)
Comment (rbramand-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c:4100]: Please use -> KERNEL_VERSION(5, 16, 0) to have same format across file

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c [hunk 6]
```diff
@@ -866,7 +866,9 @@ static void queue_req_free(struct qdma_stream_queue *queue,
 static void inline cmpl_aio(struct kiocb *kiocb, unsigned int done_bytes,
 		int error)
 {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,16,0)
```
Comment (rbramand-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c:869]: fix format in KERNEL_VERSION

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c [hunk 7]
```diff
@@ -1253,10 +1255,18 @@ static ssize_t queue_write_iter(struct kiocb *kiocb, struct iov_iter *io)
 	}
 
 	if (!is_sync_kiocb(kiocb)) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,4,0)
```
Comment (rbramand-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c:1258]: fix format in rest of files also for KERNEL_VERSION

Comment (sureshreddyavula) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c:1258]: @rbramand-xilinx I don't see the any format difference for KERNEL_VERSION with the existing code in 2021.1 branch. Can you summarize more if we are missing anything?

File: src/CMakeLists.txt (hunk 5 above)
Comment (sureshreddyavula) [reply src/CMakeLists.txt:65]: Sure

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c (hunk 7 above)
Comment (rbramand-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c:1258]: use KERNEL_VERSION(6, 4, 0)  -> I mean use space after comma's. Its trivial but good to have
